### ISSUE TITLE
Cannot use event.metadata in my projection

### ISSUE THREAD

**Issue by jokesterfr** [Sergej-Popov/event-store-projection-testing#5]
Hi!
Thank you for this repo 👍 

The `projection.processEvent()` method exposed by `util.js` does not allow to set a "metadata" object in my event. This fails executing my projection which uses it:

```
SyntaxError: Unexpected token o in JSON at position 1
        at JSON.parse (<anonymous>)
    > 53 |         doSmthg(event.data.stuff, event.metadata.correlationId);
                                                   ^
```

Do you have any clue on this?

**Comment by Sergej-Popov** [comment]
It has been a long time since I've used it. But maybe if you could give more code (projection and test) to reproduce I could take a look.

**Comment by jokesterfr** [comment]
Hi Sergej,
yes for sure, here is an example of projection:

```.js
//my.projection.js
var fromCategory =
  fromCategory || require("event-store-projection-testing").scope.fromCategory;
var emit = emit || require("event-store-projection-testing").scope.emit;
var options = options || require("event-store-projection-testing").scope.emit;

options({
  resultStreamName: "$projections-app-result",
});

function notifyApp(appId, correlationId) {
  emit(
    `app-${appId}`,
    "GreenEvent",
    {},
    {
      created_at: new Date(),
      version: 1,
      correlationId,
    }
  );
}

fromCategory("app")
  .foreachStream()
  .when({
    $init: function () {
      return {
        blued: false,
        yellowed: false,
      };
    },
    BlueEvent: function (state, event) {
      state.blued = true;
      if (state.blued && state.yellowed) {
        state.blued = false;
        state.yellowed = false;
        notifyApp(event.data.appId, event.metadata.correlationId);
      }
      return state;
    },
    YellowEvent: function (state, event) {
      state.yellowed = true;
      if (state.blued && state.yellowed) {
        notifyApp(event.data.appId, event.metadata.correlationId);
        state.blued = false;
        state.yellowed = false;
      }
      return state;
    },
  });

```

My attempt to test it:
```.js
// my.projection.spec.js
const projection = require("event-store-projection-testing");
require("./my.projection");

describe("Testing my projection", () => {
  beforeEach(() => {
    projection.initialize();
  });

  it("should emit a new GreenEvent", () => {
    projection.setState({
      blued: false,
      yellowed: false,
    });
    projection.processEvent(
      "app-17",
      "BlueEvent",
      {
        appId: 17,
      },
      {
        correlationId: "my-id",
      }
    );
    projection.processEvent(
      "app-17",
      "YellowEvent",
      {
        appId: 17,
      },
      {
        correlationId: "my-id",
      }
    );
    expect(projection.getState().blued).toBe(true);
    expect(projection.getState().yellowed).toBe(true);
  });
});
```

And the output of jest testing:

```sh
$ jest ./src/projections/my.projection.spec.js
 FAIL  src/projections/my.projection.spec.js
  Testing my projection
    ✕ should emit a new GreenEvent (2 ms)

  ● Testing my projection › should emit a new GreenEvent

    TypeError: Cannot read property 'correlationId' of null

      42 |       state.yellowed = true;
      43 |       if (state.blued && state.yellowed) {
    > 44 |         notifyApp(event.data.appId, event.metadata.correlationId);
         |                                                    ^
      45 |         state.blued = false;
      46 |         state.yellowed = false;
      47 |       }

      at YellowEvent (projections/my.projection.js:44:52)
      at callHandler (../node_modules/event-store-projection-testing/prelude/Projections.js:239:22)
      at processEvent (../node_modules/event-store-projection-testing/prelude/Projections.js:375:17)
      at Object.process_event (../node_modules/event-store-projection-testing/prelude/Projections.js:120:9)
      at util.self.processEvent (../node_modules/event-store-projection-testing/src/util.js:29:39)
      at Object.<anonymous> (projections/my.projection.spec.js:24:16)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        0.464 s
```

**Comment by Sergej-Popov** [comment]
Metadata property wasn't being passed. Fixed in 3fe3043. Released in [2.1.0](https://www.npmjs.com/package/event-store-projection-testing)

**Comment by cdesiles** [comment]
Thank you!
